mllam_data_prep/derived_variables.py

```diff
@@ -0,0 +1,459 @@
+    -------
+    field: xr.DataArray or tuple
+        The derived field
+    """
+    for attribute in ["units", "long_name"]:
+        if attribute not in field.attrs or field.attrs[attribute] is None:
+            if attribute in field_attributes.keys():
+                field.attrs[attribute] = field_attributes[attribute]
+            else:
+                # The expected attributes are empty and the attributes have not been
+                # set during the calculation of the derived variable
+                raise ValueError(
+                    f"The attribute '{attribute}' has not been set for the derived"
+                    f" variable '{field.name}' (most likely because you are using a"
+                    " function external to `mlllam-data-prep` to derive the field)."
+                    " This attribute has not been defined in the 'attributes' section"
+                    " of the config file either. Make sure that you add it to the"
+                    f" 'attributes' section of the derived variable '{field.name}'."
+                )
+        else:
+            if attribute in field_attributes.keys():
+                logger.warning(
+                    f"The attribute '{attribute}' of the derived field"
+                    f" {field.name} is being overwritten from"
+                    f" '{field.attrs[attribute]}' to"
+                    f" '{field_attributes[attribute]}' according"
+                    " to specification in the config file."
+                )
+                field.attrs[attribute] = field_attributes[attribute]
+            else:
+                # Attributes are set and nothing has been defined in the config file
+                pass
+
+    return field
+
+
+def _return_dropped_coordinates(ds_subset, ds_input, required_coordinates, chunks):
+    """Return the coordinates that have been reset."""
+    for req_coord in required_coordinates:
+        if req_coord in chunks:
+            ds_subset.coords[req_coord] = ds_input[req_coord]
+
+    return ds_subset
+
+
+def calculate_toa_radiation(lat, lon, time):
+    """
+    Function for calculating top-of-the-atmosphere radiation
+
+    Parameters
+    ----------
+    lat : xr.DataArray or float
+        Latitude values
+    lon : xr.DataArray or float
+        Longitude values
+    time : xr.DataArray or datetime object
+        Time
+
+    Returns
+    -------
+    toa_radiation: xr.DataArray or float
+        TOA radiation data
+    """
+    logger.info("Calculating top-of-atmosphere radiation")
+
+    # Solar constant
+    E0 = 1366  # W*m**-2
+
+    day = time.dt.dayofyear
+    hr_utc = time.dt.hour
+
+    # Eq. 1.6.1a in Solar Engineering of Thermal Processes 4th ed.
+    dec = np.pi / 180 * 23.45 * np.sin(2 * np.pi * (284 + day) / 365)
+
+    hr_lst = hr_utc + lon / 15
+    hr_angle = 15 * (hr_lst - 12)
+
+    # Eq. 1.6.2 with beta=0 in Solar Engineering of Thermal Processes 4th ed.
+    cos_sza = np.sin(lat * np.pi / 180) * np.sin(dec) + np.cos(
```

Comment:
Can cos_sza be defined in words in a comment or will it be too long?

Reply:
Done :)